drivers/shared/executor/executor_linux.go

```diff
@@ -339,10 +339,21 @@ func (l *LibcontainerExecutor) Shutdown(signal string, grace time.Duration) erro
 		case <-time.After(grace):
 			// Force kill all container processes after grace period,
 			// hence `true` argument.
-			return l.container.Signal(os.Kill, true)
+			if err := l.container.Signal(os.Kill, true); err != nil {
+				return err
+			}
 		}
 	} else {
-		return l.container.Signal(os.Kill, true)
+		if err := l.container.Signal(os.Kill, true); err != nil {
```

Comment:
You can simplify this code by removing the `else` block and reusing the signal block: 

```go
		//...
		select {
		case <-l.userProcExited:
			return nil
		case <-time.After(grace):
		}
	}

	// Force kill all container processes after grace period,
	// hence `true` argument.
	if err := l.container.Signal(os.Kill, true); err != nil {
		return err
	}
```